components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v2/impl/core/PasswordRecoveryService.java

```diff
@@ -0,0 +1,605 @@
+        try {
+            PasswordRecoverDTO passwordRecoverDTO = UserAccountRecoveryServiceDataHolder.getPasswordRecoveryManager()
+                    .notify(recoveryId, channelId, tenantDomain,
+                            RecoveryUtil.buildPropertiesMap(recoveryRequest.getProperties()));
+            if (passwordRecoverDTO == null) {
+                if (LOG.isDebugEnabled()) {
+                    String message = String
+                            .format("No password recovery data object for recovery code : %s", recoveryId);
+                    LOG.debug(message);
+                }
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+            }
+            return buildPasswordRecoveryResponse(tenantDomain, passwordRecoverDTO.getNotificationChannel(),
+                    passwordRecoverDTO);
+        } catch (IdentityRecoveryClientException e) {
+            if (StringUtils.isEmpty(e.getErrorCode())) {
+                throw RecoveryUtil.handleException(e.getErrorCode(), Constants.STATUS_CONFLICT_MESSAGE_DEFAULT,
+                        e.getMessage(), Response.Status.CONFLICT);
+            }
+            String errorCode = RecoveryUtil.prependOperationScenarioToErrorCode(e.getErrorCode(),
+                    IdentityRecoveryConstants.PASSWORD_RECOVERY_SCENARIO);
+            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_PASSWORD_RECOVERY_WITH_NOTIFICATIONS_NOT_ENABLED.
+                    getCode().equals(e.getErrorCode()) || IdentityRecoveryConstants.ErrorMessages.
+                    ERROR_CODE_DISABLED_ACCOUNT.getCode().equals(e.getErrorCode()) || IdentityRecoveryConstants.
+                    ErrorMessages.ERROR_CODE_LOCKED_ACCOUNT.getCode().equals(e.getErrorCode())) {
+                throw RecoveryUtil.handleException(errorCode, Constants.STATUS_FORBIDDEN_MESSAGE_DEFAULT,
+                        e.getMessage(), Response.Status.FORBIDDEN);
+            } else if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_ACCOUNT_RECOVERY_DATA.getCode().equals(
+                    e.getErrorCode())) {
+                throw RecoveryUtil.handleException(errorCode, Constants.STATUS_NOT_FOUND_MESSAGE_DEFAULT,
+                        e.getMessage(), Response.Status.NOT_FOUND);
+            } else if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_CODE.getCode().equals(
+                    e.getErrorCode()) || IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_RECOVERY_CODE
+                    .getCode().equals(e.getErrorCode())) {
+                throw RecoveryUtil.handleException(errorCode, Constants.STATUS_METHOD_NOT_ACCEPTED_MESSAGE_DEFAULT,
+                        e.getMessage(), Response.Status.NOT_ACCEPTABLE);
+            } else {
+                throw RecoveryUtil.handleException(errorCode, Constants.STATUS_CONFLICT_MESSAGE_DEFAULT,
+                        e.getMessage(), Response.Status.CONFLICT);
```

Comment:
These lines are duplicated with code lines https://github.com/wso2/identity-api-user/pull/179/files#diff-4d19de3b2c8b95acae1c7574c361ca6a3eb54a46f9a326124e746106c6a10cafR164-R175  (Exception handling of recoverUsername)

Reply:
Fixed with 089d7d6472464375905b2d59aff6abdbd8caec21